homeassistant/helpers/script.py

```diff
@@ -1024,6 +1028,27 @@ async def async_run_with_trace(idx: int, script: Script) -> None:
             if isinstance(result, Exception):
                 raise result
 
+    async def _async_data_sources_step(self):
+        """Handle data sources."""
+        self._step_log("data sources")
+        data_sources = self._action[CONF_DATA_SOURCES]
+        tasks = []
+        for _, config in data_sources.items():
```

Comment:
This should be done in parallel using an `asyncio.gather`

Reply:
Hmm no, that would not allow us to use variables from previous data sources. We don't do that for templates either.

Reply:
But that does mean that we should assign the results to `self._variables` before executing the next item.

Reply:
Oh I see you're doing it in gather already. Not sure which one is better 🤷

Reply:
Setting variables for reuse in follow up stages makes sense to me, and i think the code is simpler. I've switched to do it that way.